File: contests/lc/3577.CounttheNumberofComputerUnlockingPermutations/solution.go
package main

const mod int64 = 1_000_000_007

// countUnlockingPermutations returns the number of valid unlocking orders.
// 0 must be the unique minimum complexity; otherwise some node cannot be unlocked.
func countUnlockingPermutations(complexity []int) int {
	n := len(complexity)
	root := complexity[0]
	for i := 1; i < n; i++ {
		if complexity[i] <= root {
			return 0
		}
	}

	// Any order of the remaining (n-1) computers works.
	fact := int64(1)
	for i := 2; i <= n-1; i++ {
		fact = fact * int64(i) % mod
	}
	return int(fact)
}

File: contests/lc/3577.CounttheNumberofComputerUnlockingPermutations/solution.py
from typing import List


MOD = 1_000_000_007


class Solution:
    def countPermutations(self, complexity: List[int]) -> int:
        root = complexity[0]
        for value in complexity[1:]:
            if value <= root:
                return 0

        # Factorial of (n - 1) modulo MOD.
        fact = 1
        for x in range(2, len(complexity)):
            fact = (fact * x) % MOD
        return fact
